legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/TestableCommandsSupport.java

```diff
@@ -71,79 +62,122 @@ public final class TestableCommandsSupport implements CommandsSupport
+    public Optional<LegendTest> collectTests(PackageableElement element)
     {
         if (this.isTestable(element))
         {
             // todo enhance engine to provide test suites per testables
             List<? extends TestSuite> testSuites = this.extension.getTestSuites(element);
             if (!testSuites.isEmpty())
             {
-                consumer.accept(RUN_TESTS_COMMAND_ID, RUN_TESTS_COMMAND_TITLE, element.sourceInformation);
-                testSuites.forEach(testSuite ->
+                List<LegendTest> testSuitesLegendTest = testSuites.stream().map(testSuite ->
                 {
-                    consumer.accept(RUN_TEST_SUITE_COMMAND_ID, RUN_TEST_SUITE_COMMAND_TITLE, testSuite.sourceInformation, Collections.singletonMap(TEST_SUITE_ID, testSuite.id));
-                    testSuite.tests.forEach(test -> consumer.accept(RUN_TEST_COMMAND_ID, RUN_TEST_COMMAND_TITLE, test.sourceInformation, Maps.fixedSize.with(TEST_SUITE_ID, testSuite.id, TEST_ID, test.id)));
-                });
+                    List<LegendTest> testLegendTest = testSuite.tests.stream().map(test -> LegendTest.newLegendTest(
+                            List.of(),
+                            SourceInformationUtil.toLocation(test.sourceInformation),
+                            element.getPath(),
+                            testSuite.id,
+                            test.id
+                    )).collect(Collectors.toList());
+
+                    return LegendTest.newLegendTest(
+                            testLegendTest,
+                            SourceInformationUtil.toLocation(testSuite.sourceInformation),
+                            element.getPath(),
+                            testSuite.id
+                    );
+                }).collect(Collectors.toList());
+
+                return Optional.of(
+                        LegendTest.newLegendTest(
+                                testSuitesLegendTest,
+                                SourceInformationUtil.toLocation(element.sourceInformation),
+                                element.getPath()
+                        )
+                );
             }
         }
+
+        return Optional.empty();
     }
 
-    @Override
-    public Iterable<? extends LegendExecutionResult> executeCommand(SectionState section, PackageableElement element, String commandId, Map<String, String> executableArgs, Map<String, Object> inputParameters)
+    public List<LegendTestExecutionResult> executeTests(SectionState sectionState, PackageableElement element, String testId, Set<String> excludedTestIds)
     {
-        TextLocation location = SourceInformationUtil.toLocation(element.sourceInformation);
-        String entityPath = element.getPath();
-        switch (commandId)
+        return this.collectTests(element)
+                .map(test ->
+                {
+                    LinkedList<LegendTest> q = new LinkedList<>();
```

Comment:
I think ArrayDeque is a better queue implementation than LinkedList.
```suggestion
                    Deque<LegendTest> q = new ArrayDeque<>();
```

Reply:
Fixed!